jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -146,21 +143,21 @@ YUI.add('user-profile', function() {
       @param {Object} data The data from the request.
     */
     _fetchEntitiesCallback: function(type, error, data) {
+      // Turn off the loader, regardless of success or error.
+      if (type === 'charm') {
```

Comment:
Yes, that is the reason why. If there is an error, we still want the load indicator to go away. Which is why I put a comment to that effect on line 146 ;-)